core/server/models/base/index.js

```diff
@@ -717,20 +718,20 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
         }
 
         return itemCollection.fetchPage(options).then(function formatResponse(response) {
-            var data = {},
-                models;
-
-            options.columns = requestedColumns;
-            models = response.collection.toJSON(options);
+            // Attributes are being filtered here, so they are not leaked into calling layer
+            // where models are serialized to json and do not do more filtering
+            const data = response.collection.models.map((model) => {
+                if (requestedColumns) {
+                    model.attributes = _.pick(model.attributes, requestedColumns);
```

Comment:
While adding the filtering comment remembered I had a question about the `requestedColumns` variable here. It's used to be used in a weird way - like [this](https://github.com/TryGhost/Ghost/pull/9899/files#diff-f8d02ad12b13d44b912bbf46cf7193b6L723). And I'm not sure if it was meant to be called with originally requested columns or we want to call the pick with `options.columns` instead maybe?

Reply:
We def want to use the original columns.

But

> options.columns = requestedColumns;

Is no longer needed, because earlier days of Ghost, the model layer modified the original options. We now create a clone to ensure we don't modify the passed options.